src/kinetics/InterfaceKinetics.cpp

```diff
@@ -953,4 +953,23 @@ void InterfaceKinetics::applyStickingCorrection(double T, double* kf)
     }
 }
 
+double InterfaceKinetics::rxn_enthalpy(const Composition& reactants, const Composition& products)
+{
+    for (size_t n = 0; n < nPhases(); n++) {
+        thermo(n).getPartialMolarEnthalpies(m_grt.data() + m_start[n]);
+    }
+    double rxn_deltaH = 0;
+    for (auto iter = products.begin(); iter != products.end(); ++iter) {
+        int k = kineticsSpeciesIndex(iter->first);
+        double stoich = iter->second;
+        rxn_deltaH += m_grt.data()[k] * stoich;
+    }
```

Comment:
A few suggestions here, which also apply to the loop below.
```suggestion
    for (const auto& product : products) {
        size_t k = kineticsSpeciesIndex(product.first);
        double stoich = product.second;
        rxn_deltaH += m_grt[k] * stoich;
    }
```